### PR TITLE
Herons : Changed the signature of the function

### PR DESCRIPTION
**Pull request by satyabarghav**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

I have already raised this issue [here](https://github.com/TheAlgorithms/Java/issues/4685#issue-1929808965) 
The required changes were made to make it work with even larger set of data types like double and float and further documentation was also added ro ease the readability and understandability of the code.

### CODE REVIEW

**Comment by satyabarghav** [comment]
fix #4685

**Comment by satyabarghav** [comment]
#4685

File: src/main/java/com/thealgorithms/maths/HeronsFormula.java [hunk 1]
```diff
@@ -1,18 +1,26 @@
 package com.thealgorithms.maths;
 
 /**
+ * Wikipedia for HeronsFormula => https://en.wikipedia.org/wiki/Heron%27s_formula
  * Find the area of a triangle using only side lengths
  */
 
 public class HeronsFormula {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/maths/HeronsFormula.java:8]: We can make it a proper _utility class_:

```suggestion
public final class HeronsFormula {
    private HeronsFormula() {
    }
```

File: src/main/java/com/thealgorithms/maths/HeronsFormula.java [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
-        area = Math.sqrt((s) * (s - a) * (s - b) * (s - c));
-        return area;
+    /*
+     * A function to get the Area of a Triangle using Heron's Formula
+     * @param s1,s2,s3 => the three sides of the Triangle
+     * @return area using the formula (√(s(s – s1)(s – s2)(s – s3)))
+     * here s is called semi-perimeter and it is the half of the perimeter (i.e; s = (s1+s2+s3)/2)
+     * @author satyabarghav
+     */
+    public static double Herons(double a, double b, double c) {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/maths/HeronsFormula.java:16]:
```suggestion
    public static double Herons(final double a, final double b, final double c) {
```

File: src/main/java/com/thealgorithms/maths/HeronsFormula.java [hunk 3]
```diff
@@ -1,18 +1,26 @@
...
+        if (!(a + b > c && b + c > a && c + a > b)) {
+            throw new IllegalArgumentException("Triangle can't be formed with the given side lengths");
```
Comment (vil02) [inline src/main/java/com/thealgorithms/maths/HeronsFormula.java:18]: Could you please add a test _covering_ this line? (and all of the cases in the `if` statement above)?

File: src/main/java/com/thealgorithms/maths/HeronsFormula.java [hunk 4]
```diff
@@ -1,18 +1,26 @@
...
+        } else {
+            double s = (a + b + c) / 2.0;
+            double area = 0;
+            area = Math.sqrt((s) * (s - a) * (s - b) * (s - c));
+            return area;
+        }
```
Comment (vil02) [inline src/main/java/com/thealgorithms/maths/HeronsFormula.java:24]: What do you think about the following simplification?

```suggestion
        }
        final double s = (a + b + c) / 2.0;
        return Math.sqrt((s) * (s - a) * (s - b) * (s - c));
```

**Review by vil02 — CHANGES_REQUESTED** [review]

**Comment by vil02** [comment]
Fixes #4685.

File: src/main/java/com/thealgorithms/maths/HeronsFormula.java [hunk 5]
```diff
@@ -1,18 +1,27 @@
...
+     * A function to get the Area of a Triangle using Heron's Formula
+     * @param s1,s2,s3 => the three sides of the Triangle
+     * @return area using the formula (√(s(s – s1)(s – s2)(s – s3)))
+     * here s is called semi-perimeter and it is the half of the perimeter (i.e; s = (s1+s2+s3)/2)
+     * @author satyabarghav
+     */
+    private HeronsFormula() {
+    }
+    public static double Herons(final double a, final double b, final double c) {
+        if (!(a + b > c && b + c > a && c + a > b)) {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/maths/HeronsFormula.java:20]: What do you think about changing the inequalities to `>=`? I think there is nothing wrong in a triangle with side lengths 1, 2, 3 (i.e. it has are 0). Moreover, I would suggest to add a check if the provided side lengths are `>= 0`.

```suggestion
        if (!(a + b >= c && b + c >= a && c + a >= b)) {
```

File: pom.xml [hunk 6]
```diff
@@ -28,6 +28,12 @@
     </dependencyManagement>
 
     <dependencies>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>5.8.2</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (vil02) [inline pom.xml:36]: Why is this needed?

File: src/main/java/com/thealgorithms/maths/HeronsFormula.java [hunk 7]
```diff
@@ -1,18 +1,27 @@
...
+            throw new IllegalArgumentException("Triangle can't be formed with the given side lengths");
+        } else {
+            final double s = (a + b + c) / 2.0;
+            return Math.sqrt((s) * (s - a) * (s - b) * (s - c));
+        }
```
Comment (vil02) [inline src/main/java/com/thealgorithms/maths/HeronsFormula.java:25]:
```suggestion
        }
        final double s = (a + b + c) / 2.0;
        return Math.sqrt((s) * (s - a) * (s - b) * (s - c));
```

File: src/test/java/com/thealgorithms/maths/HeronsFormulaTest.java [hunk 8]
```diff
@@ -1,7 +1,6 @@
 package com.thealgorithms.maths;
 
-import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.*;
```
Comment (vil02) [inline src/test/java/com/thealgorithms/maths/HeronsFormulaTest.java:3]: I would suggest to avoid `*`-like imports.

**Comment by satyabarghav** [comment]
Made the requested changes, I request the Reviewer(s) to please tag this as hacktoberfest-accepted.

**Review by vil02 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 6 above)
Comment (satyabarghav) [reply pom.xml:36]: Removed it

File: src/main/java/com/thealgorithms/maths/HeronsFormula.java (hunk 5 above)
Comment (satyabarghav) [reply src/main/java/com/thealgorithms/maths/HeronsFormula.java:20]: I don't think that's correct according to the definition and a triangle with sides 1,2,3 cannot be constructed Geometrically
image

File: src/main/java/com/thealgorithms/maths/HeronsFormula.java [hunk 9]
```diff
@@ -1,18 +1,27 @@
...
+    /*
+     * A function to get the Area of a Triangle using Heron's Formula
+     * @param s1,s2,s3 => the three sides of the Triangle
+     * @return area using the formula (√(s(s – s1)(s – s2)(s – s3)))
+     * here s is called semi-perimeter and it is the half of the perimeter (i.e; s = (s1+s2+s3)/2)
+     * @author satyabarghav
+     */
+    private HeronsFormula() {
+    }
+    public static double Herons(final double a, final double b, final double c) {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/maths/HeronsFormula.java:19]: Using this opportunity, I would suggest to rename the method to start with the _small_ `h`.

```suggestion
    public static double herons(final double a, final double b, final double c) {
```

File: src/main/java/com/thealgorithms/maths/HeronsFormula.java (hunk 5 above)
Comment (vil02) [inline src/main/java/com/thealgorithms/maths/HeronsFormula.java:20]: What do you think about adding a check if provided side lengths are positive?

File: src/test/java/com/thealgorithms/maths/HeronsFormulaTest.java [hunk 10]
```diff
@@ -24,4 +24,9 @@ void test3() {
     void test4() {
         Assertions.assertEquals(HeronsFormula.Herons(4, 5, 8), 8.181534085976786);
     }
+
+    @Test
+    public void testCalculateAreaWithInvalidInput() {
+        Assertions.assertThrows(IllegalArgumentException.class, () -> { HeronsFormula.Herons(1, 2, 3); });
```
Comment (vil02) [inline src/test/java/com/thealgorithms/maths/HeronsFormulaTest.java:30]: This will give _100%_ coverage.

```suggestion
        Assertions.assertThrows(IllegalArgumentException.class, () -> { HeronsFormula.Herons(1, 2, 3); });
        Assertions.assertThrows(IllegalArgumentException.class, () -> { HeronsFormula.Herons(2, 1, 3); });
        Assertions.assertThrows(IllegalArgumentException.class, () -> { HeronsFormula.Herons(3, 2, 1); });
```

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/maths/HeronsFormula.java (hunk 7 above)
Comment (vil02) [inline src/main/java/com/thealgorithms/maths/HeronsFormula.java:25]: This `else` block is not needed.

```suggestion
        }
        final double s = (a + b + c) / 2.0;
        return Math.sqrt((s) * (s - a) * (s - b) * (s - c));
```

File: src/main/java/com/thealgorithms/maths/HeronsFormula.java (hunk 5 above)
Comment (vil02) [reply src/main/java/com/thealgorithms/maths/HeronsFormula.java:20]: A _"1, 2, 3"_-triangle would be empty. I would not exclude it. But I understand that it might be a bit controversial. 

Please add a check that provided side lengths are all positive.

**Comment by vil02** [comment]
> Made the requested changes, I request the Reviewer(s) to please tag this as hacktoberfest-accepted.

This is not needed, since this repository has `hacktoberfest` _tag_. The [website](https://hacktoberfest.com) will recognize your contribution.

File: src/main/java/com/thealgorithms/maths/HeronsFormula.java (hunk 5 above)
Comment (satyabarghav) [reply src/main/java/com/thealgorithms/maths/HeronsFormula.java:20]: I think it's fine, Added it.

**Comment by vil02** [comment]
@satyabarghav there was a subtle error in the check if all of the provided sidelights are positive, namely, the tests:
```java
Assertions.assertThrows(IllegalArgumentException.class, () -> { HeronsFormula.herons(1, 1, 0); });
Assertions.assertThrows(IllegalArgumentException.class, () -> { HeronsFormula.herons(1, 0, 1); });
Assertions.assertThrows(IllegalArgumentException.class, () -> { HeronsFormula.herons(0, 1, 1); });
```
did not pass.

Always add tests.

To speed up the whole process I allowed myself to push a commit into your branch - I hope you do not mind.

**Review by vil02 — APPROVED** [review]

**Comment by satyabarghav** [comment]
Cool, thanks for being with me :D
